### PR TITLE
Send mail to user after 8 hours if user have uncompleted test

### PR DESCRIPTION
**Pull request by Dat1606**

### CODE REVIEW

**Comment by Dat1606** [comment]
https://edu-redmine.framgia.vn/issues/2153?issue_count=22&issue_position=1&next_issue_id=2006

**Comment by Dat1606** [comment]
ready

File: app/models/user_lesson.rb [hunk 1]
```diff
@@ -11,4 +12,10 @@ def self.find_by_user user_id
   def self.collect_id_by_user user_id
     where("user_id = ?", user_id).pluck(:id)
   end
+
+  def send_mail
+    if self.results.count < self.lesson.vocabularies.count
+      UncompletedTestJob.set(wait: 8.hours).perform_later(self)
```
Comment (huongnguyenmta) [inline app/models/user_lesson.rb]: move `8` vào settings em nhá

File: config/routes.rb [hunk 2]
```diff
@@ -12,6 +12,7 @@
   resources :user_lessons
   resources :results, only: [:create, :destroy]
   resources :users
+  default_url_options host: "localhost:3000"
```
Comment (huongnguyenmta) [inline config/routes.rb]: set theo từng môi trường em nhá (development, production)

File: config/environments/development.rb [hunk 3]
```diff
@@ -9,11 +9,20 @@
   # Do not eager load code on boot.
   config.eager_load = false
 
+  config.action_mailer.default_url_options = { :host => "localhost:3000" }
```
Comment (huongnguyenmta) [inline config/environments/development.rb]: thêm config luôn trong môi trường production ngen em

**Comment by huongnguyenmta** [comment]
commented

**Comment by Dat1606** [comment]
ready

**Pull request merged**